var/spack/repos/builtin/packages/migrate/package.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Migrate(AutotoolsPackage):
+    """Migrate estimates effective population sizes and past migration rates
+       between n population assuming a migration matrix model with asymmetric
+       migration rates and different subpopulation sizes"""
+
+    homepage = "http://popgen.sc.fsu.edu/"
+    url      = "http://popgen.sc.fsu.edu/currentversions/migrate-3.6.11.src.tar.gz"
+
+    version('3.6.11', sha256='a9ba06a4e995a45b8d04037f5f2da23e1fe64a2f3565189bdd50c62c6fe01fb8')
+
+    variant('mpi', default=False,
+            description='Build MPI binaries')
+
+    depends_on('autoconf')
+    depends_on('automake')
+    depends_on('libtool')
+    depends_on('m4')
+
+    depends_on('openmpi', type=('build', 'link', 'run'), when='+mpi')
+
+    configure_directory = 'src'
+
+    def build(self, spec, prefix):
+        with working_dir('src'):
+            # this software is written with parts both in C and C++.
+            # it tries to link using gcc which causes problems, so this package
+            # explicitly links with g++ (CXX) instead.
```

Comment:
wouldn't the easier solution be to override `CC` to also have the content of `CXX`?

Reply:
There are some portions of code which are strictly C and produce errors under g++